builder/builder-manifest.c

```diff
@@ -2174,3 +2186,170 @@ builder_manifest_run (BuilderManifest *self,
+    {
+      self->appstream_compose = overrides->appstream_compose;
+    }
+  if (overrides->sdk_extensions)
+    {
+      g_strfreev (self->sdk_extensions);
+      self->sdk_extensions = g_strdupv (overrides->sdk_extensions);
+    }
+  if (overrides->platform_extensions)
+    {
+      g_strfreev (self->platform_extensions);
+      self->platform_extensions = g_strdupv (overrides->platform_extensions);
+    }
+  if (overrides->tags)
+    {
+      g_strfreev (self->tags);
+      self->tags = g_strdupv (overrides->tags);
+    }
+  if (overrides->cleanup)
+    {
+      g_strfreev (self->cleanup);
+      self->cleanup = g_strdupv (overrides->cleanup);
+    }
+  if (overrides->cleanup_commands)
+    {
+      g_strfreev (self->cleanup_commands);
+      self->cleanup_commands = g_strdupv (overrides->cleanup_commands);
+    }
+  if (overrides->cleanup_platform)
+    {
+      g_strfreev (self->cleanup_platform);
+      self->cleanup_platform = g_strdupv (overrides->cleanup_platform);
+    }
+  if (overrides->finish_args)
+    {
+      g_strfreev (self->finish_args);
+      self->finish_args = g_strdupv (overrides->finish_args);
+    }
+  if (overrides->rename_desktop_file)
+    {
+      g_free (self->rename_desktop_file);
+      self->rename_desktop_file = g_strdup (overrides->rename_desktop_file);
+    }
+  if (overrides->rename_appdata_file)
+    {
+      g_free (self->rename_appdata_file);
+      self->rename_appdata_file = g_strdup (overrides->rename_appdata_file);
+    }
+  if (overrides->rename_icon)
+    {
+      g_free (self->rename_icon);
+      self->rename_icon = g_strdup (overrides->rename_icon);
+    }
+  if (overrides->copy_icon)
+    {
+      self->copy_icon = overrides->copy_icon;
+    }
+  if (overrides->desktop_file_name_prefix)
+    {
+      g_free (self->desktop_file_name_prefix);
+      self->desktop_file_name_prefix = g_strdup (overrides->desktop_file_name_prefix);
+    }
+  if (overrides->desktop_file_name_suffix)
+    {
+      g_free (self->desktop_file_name_suffix);
+      self->desktop_file_name_suffix = g_strdup (overrides->desktop_file_name_suffix);
+    }
+
+  if (overrides->build_options)
+    {
+      if (self->build_options)
+        builder_options_apply_overrides (self->build_options, overrides->build_options);
+      else
+        self->build_options = g_object_ref (overrides->build_options);
+    }
+
+  if (overrides->modules)
+    {
+      GList *l;
+      GList *l2;
+
+      expand_modules (self, NULL);
+      expand_modules (overrides, NULL);
+
+      for (l = overrides->expanded_modules; l != NULL; l = l->next)
+        {
+          BuilderModule *m = l->data;
+          for (l2 = self->expanded_modules; l2 != NULL; l2 = l2->next)
+            {
+              BuilderModule *m2 = l2->data;
+              if (!g_strcmp0 (builder_module_get_name (m), builder_module_get_name (m2)))
+                {
+                  builder_module_apply_overrides (m2, m);
+                  break;
+                }
+            }
```

Comment:
We should probably error out if an overriding module isn't found in the regular manifest.